lnwallet/rpcwallet/rpcwallet.go

```diff
@@ -1030,6 +1031,50 @@ func (r *RPCKeyRing) remoteSign(tx *wire.MsgTx, signDesc *input.SignDescriptor,
 		// If this is a BIP0086 key spend then the tap tweak is empty,
 		// otherwise it's set to the Taproot root hash.
 		in.TaprootMerkleRoot = signDesc.TapTweak
+
+	case input.SegWitV1ScriptSpendSignMethod:
+		// The script spend path is a bit more involved when doing it
+		// through the PSBT method. We need to specify the leaf hash
+		// that the signer should sign for.
+		leaf := txscript.TapLeaf{
+			LeafVersion: txscript.BaseLeafVersion,
+			Script:      signDesc.WitnessScript,
+		}
+		leafHash := leaf.TapHash()
+
+		d := in.Bip32Derivation[0]
+		in.TaprootBip32Derivation = []*psbt.TaprootBip32Derivation{{
+			XOnlyPubKey:          d.PubKey[1:],
+			LeafHashes:           [][]byte{leafHash[:]},
+			MasterKeyFingerprint: d.MasterKeyFingerprint,
+			Bip32Path:            d.Bip32Path,
+		}}
+
+		// We also need to supply a control block. But because we don't
```

Comment:
Was curious about this before reviewing the PR: is constructing the final control block a new duty of the "combiner" role?

Reply:
No, the control block needs to be known by the creator/updater of the PSBT because it is the _key_ of a `PSBT_IN_TAP_LEAF_SCRIPT` element. So in order to have any script path signed by a signer, you need to specify a (formally) valid control block, even if none of that information actually makes it into the sighash itself.

This was a bit surprising to me as well. But I guess this makes sure that all the information for the final witness stack is there after the signer added their signature and a finalizer can go ahead and assemble it.